server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2389,18 +2389,27 @@ public Pool setPoolQuantity(Pool pool, long set) {
 
     @Override
     public void regenerateDirtyEntitlements(Consumer consumer) {
-        this.regenerateDirtyEntitlements(this.entitlementCurator.listDirty(consumer));
+        if (consumer != null) {
```

Comment:
The case where I have no idea what the code path looks like, but still don't want an NPE out of left field.

I suppose I could make it an error case. Generally with a lot of the existing code paths, I tend to avoid introducing new errors if I'm not familiar with all of the cases where we use it.